ps_democqrshooksusage.php

```diff
@@ -0,0 +1,337 @@
+    }
+
+    /**
+     * @param array $params
+     *
+     * @throws CustomerException
+     */
+    private function updateCustomerReviewStatus(array $params)
+    {
+        //todo: a better would be to grab the data from array?
+
+        $customerId = $params['id'];
+        /** @var Request $request */
+        $request = $params['request'];
+        /** @var array $customerFormData */
+        $customerFormData = $request->get('customer');
+        $isAllowedForReview = (bool) $customerFormData['is_allowed_for_review'];
+
+        /** @var CommandBusInterface $commandBus */
+        $commandBus = $this->get('prestashop.core.command_bus');
+
+        try {
+            $commandBus->handle(new UpdateIsAllowedToReviewCommand(
+                $customerId,
+                $isAllowedForReview
+            ));
+        } catch (ReviewerException $exception) {
+            $this->handleException($exception);
+        }
+    }
+
+    /**
+     * Installs sample tables required for demonstration.
+     *
+     * @return bool
+     */
+    private function installTables()
+    {
+        $sql = '
+            CREATE TABLE IF NOT EXISTS `' . pSQL(_DB_PREFIX_) . 'democqrshooksusage_reviewer` (
+                `id_reviewer` INT(10) UNSIGNED NOT NULL AUTO_INCREMENT,
+                `id_customer` INT(10) UNSIGNED NOT NULL,
+                `is_allowed_for_review` TINYINT(1) NOT NULL,
+                PRIMARY KEY (`id_reviewer`)
+            ) ENGINE=' . pSQL(_MYSQL_ENGINE_) . ' COLLATE=utf8_unicode_ci;
+        ';
+
+        return Db::getInstance()->execute($sql);
+    }
+
+    /**
+     * Uninstalls sample tables required for demonstration.
+     *
+     * @return bool
+     */
+    private function uninstallTables()
+    {
+        $sql = 'DROP TABLE IF EXISTS `' . pSQL(_DB_PREFIX_) . 'democqrshooksusage_reviewer`';
+
+        return Db::getInstance()->execute($sql);
+    }
+
+    /**
+     * Handles exceptions and displays message in more user friendly form.
+     *
+     * @param ReviewerException $exception
+     */
+    private function handleException(ReviewerException $exception)
+    {
+        $exceptionDictionary = [
+            CannotCreateReviewerException::class => $this->getTranslator()->trans(
+                'Failed to create a record for customer',
+                [],
+                'Modules.Ps_DemoCQRSHooksUsage'
+            ),
+            CannotToggleAllowedToReviewStatusException::class => $this->getTranslator()->trans(
+                'Failed to toggle is allowed to review status',
+                [],
+                'Modules.Ps_DemoCQRSHooksUsage'
+            ),
+        ];
+
+        $exceptionType = get_class($exception);
+
+        /** @var FlashBagInterface $flashBag */
+        $flashBag = $this->get('session')->getFlashBag();
+
+
+        if (isset($exceptionDictionary[$exceptionType])) {
+            $flashBag->add('error', $exceptionDictionary[$exceptionType]);
```

Comment:
Nice 👍

Reply:
its nice but it caused a lot of argues here https://github.com/PrestaShop/PrestaShop/issues/13490. You can check my screenshots there to see why there are many opinions on using this - as I see we not yet decided how to use it :thinking:

Reply:
Hoho indeed that is a heated debate 😃 🔥 

Let's see how it clears out ^^ indeed